library/src/main/java/com/pokegoapi/api/player/PlayerProfile.java

```diff
@@ -77,55 +77,91 @@
 
 	/**
 	 * @param api the api
-	 * @throws LoginFailedException  when the auth is invalid
-	 * @throws RemoteServerException when the server is down/having issues
 	 */
-	public PlayerProfile(PokemonGo api) throws LoginFailedException, RemoteServerException {
+	public PlayerProfile(PokemonGo api) {
 		this.api = api;
 		this.playerLocale = new PlayerLocale();
 
 		if (playerData == null) {
-			updateProfile();
+			initProfile();
 		}
 	}
 
 	/**
-	 * Updates the player profile with the latest data.
-	 *
-	 * @throws LoginFailedException  when the auth is invalid
-	 * @throws RemoteServerException when the server is down/having issues
+	 * Init the player profile.
 	 */
-	public void updateProfile() throws RemoteServerException, LoginFailedException {
+	private void initProfile() {
 		GetPlayerMessage getPlayerReqMsg = GetPlayerMessage.newBuilder()
 				.setPlayerLocale(playerLocale.getPlayerLocale())
 				.build();
 
-		ServerRequest getPlayerServerRequest = new ServerRequest(RequestType.GET_PLAYER, getPlayerReqMsg);
-		api.getRequestHandler().sendServerRequests(
-				CommonRequest.appendCheckChallenge(getPlayerServerRequest));
-
-		try {
-			updateProfile(GetPlayerResponse.parseFrom(getPlayerServerRequest.getData()));
-		} catch (InvalidProtocolBufferException e) {
-			throw new RemoteServerException(e);
-		}
+		new AsyncServerRequest(RequestType.GET_PLAYER, getPlayerReqMsg,
+				new PokeAFunc<GetPlayerResponse, Void>() {
+					@Override
+					public Void exec(GetPlayerResponse response) {
+						parseData(response.getPlayerData());
+
+						ArrayList<TutorialStateOuterClass.TutorialState> tutorialStates =
+								getTutorialState().getTutorialStates();
+						if (tutorialStates.isEmpty()) {
+							activateAccount();
+							return null;
+						}
+
+						if (!tutorialStates.contains(TutorialStateOuterClass.TutorialState.AVATAR_SELECTION)) {
+							setupAvatar();
+							return null;
+						}
+
+						if (!tutorialStates.contains(TutorialStateOuterClass.TutorialState.POKEMON_CAPTURE)) {
+							encounterTutorialComplete();
+							return null;
+						}
+
+						if (!tutorialStates.contains(TutorialStateOuterClass.TutorialState.NAME_SELECTION)) {
+							claimCodeName();
+							return null;
+						}
+
+						if (!tutorialStates.contains(
+								TutorialStateOuterClass.TutorialState.FIRST_TIME_EXPERIENCE_COMPLETE)) {
+							firstTimeExperienceComplete();
+						}
+
+						return null;
```

Comment:
Inner thing always returns null, so why return null in ever if, and not use an if/else?